### PR TITLE
[SPARK-29899][SQL] Recursively load data in Hive table via TBLPROPERTIES

### PR DESCRIPTION
**Pull request by LantaoJin**
### What changes were proposed in this pull request?
SPARK-27990 (#24830) provide a way to recursively load data from datasource. In SQL, when query a hive table, this property passed by the `relation.tableMeta.properties`. But it is filtered out now. So we can not lookup file recursively for a Hive table.

In this PR, I don't add a new property or feature. The property `recursiveFileLookup` in TBLPROPERTIES should work in current implementation. But it's filtered out bugly.
```sql
CREATE TABLE test1 (id bigint)
STORED AS PARQUET LOCATION '$baseDir'
TBLPROPERTIES (
'recursiveFileLookup'='true')
```

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
Add an UT

### CODE REVIEW

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveMetastoreCatalogSuite.scala [hunk 1]
```diff
@@ -358,4 +358,30 @@ class DataSourceWithHiveMetastoreCatalogSuite
...
+      val innerDir = s"$baseDir/path2/path3"
+      spark.range(3).selectExpr("id").write.parquet(innerDir)
+      withTable("test1", "test2") {
+        withSQLConf(HiveUtils.CONVERT_METASTORE_PARQUET.key -> "true") {
+          spark.sql(
+            s"""
+               |CREATE TABLE test1 (id bigint)
+               |STORED AS PARQUET LOCATION '$baseDir'
+               |TBLPROPERTIES (
+               | 'recursiveFileLookup'='true')
```
Comment (nchammas) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveMetastoreCatalogSuite.scala]: Sorry to ask tangential questions, but I'm curious: Will the Metastore track this property somehow? i.e. If I create a table with `'recursiveFileLookup'='true'` using Spark, can I query it from Presto and see the same data, provided that both are pointed at the same Metastore? Will the Metastore just track the table property, or will it also track the list of data paths that were detected when the table was created or refreshed?

**Comment by cloud-fan** [comment]
can you describe the expected behavior? To me, the hive metastore already tells the directory structure: if it's partitioned, then data files are under each partition directory. Otherwise, data files are under table directory. Why do we need to lookup files recursively?

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveMetastoreCatalogSuite.scala (hunk 1 above)
Comment (LantaoJin) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveMetastoreCatalogSuite.scala]: Thanks to point me this. Maybe 'spark.recursiveFileLookup' is much more meaningful for user.

**Comment by LantaoJin** [comment]
@cloud-fan The reason is very simple but I am not sure it's correct for Hive:
We found some data source paths of hive table are nested. And I found a way to handle this in Spark datasource (#24830). Seems datasource API has some reasons to load data recursively. So I think table might have the same approach. I can close this since the issue can be fixed by removing nested paths if this patch looks unreasonable.

**Comment by cloud-fan** [comment]
load files recursively may make sense to some data sources but not tables. We have a clear policy about the filed layout for tables. Please close this.

**Comment by LantaoJin** [comment]
@cloud-fan thanks for pointing this. close

**Pull request closed without merging**